src/result.rs

```diff
@@ -7,7 +7,7 @@
 
 use crate::iter::plumbing::*;
 use crate::iter::*;
-use std::sync::Mutex;
+use rayon_core::sync::Mutex;
```

Comment:
This case only ever uses `try_lock` -- is that still a problem for the web main?

Reply:
Ah, no, I think that should be fine.

Reply:
Do you think we could change `iter/par_bridge.rs` to use `try_lock` instead of `lock` as well? It already uses custom loop, so functionally it would be equivalent, but I'm not sure how common it is for `try_lock` to fail there and, thus, whether such spinning would cause performance regression on other platforms.

If we could, then we could avoid dependency on `wasm_sync` in `rayon` altogether and keep it only in `rayon-core`.

Reply:
@cuviper Ping - which route would you prefer - should I change it to `try_lock` or add `wasm_sync` dependency to `rayon` as well?

Reply:
Spinning on `try_lock` would regress #795, so let's not go there.

The `par_bridge` `lock` should normally only be called from within the pool, in which case `wasm_sync` will properly block instead of spinning, right? However, it's also possible to make the main thread part of the pool since #1063. So either we need to let `wasm_sync` spin there, or we need to document that wasm users shouldn't do that when they have real threading available.

Anyway, it's not a big deal to add the dependency to `rayon` when it's already forced by `rayon-core`.

Reply:
Fair enough. Based on this, sounds like adding `wasm_sync` to `rayon` is indeed a better option for cross-platform compat. I'll do that.

Reply:
Done.